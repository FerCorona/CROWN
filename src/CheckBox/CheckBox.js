import React from 'react';

import { v4 as uuidv4 } from 'uuid';

import Label from '../Label';

const CheckBox = ({
  id,
  text,
  onChange,
  isChecked,
  stateName,
  extraStyle
}) => (
  <div className={`CROWNCheckBox ${extraStyle}`}>
    <input
      id={id || uuidv4()}
      type='checkbox'
      className='Active'
      onChange={() => onChange({
        stateName,
        value: !isChecked
      })}
      checked={isChecked}
    />
    {
      text && (
        <div className='Label'>
          <Label size='small' text={text} color='--black-1' />
        </div>
      )
    }
  </div>
);

CheckBox.defaultProps = {
  text: null,
  isChecked: false,
  onChange: () => { },
  extraStyle: ''
};

export default CheckBox;