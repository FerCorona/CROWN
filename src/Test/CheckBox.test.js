import React from 'react';
import renderer from 'react-test-renderer';
import CheckBox from '../CheckBox/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <CheckBox id={1234} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});