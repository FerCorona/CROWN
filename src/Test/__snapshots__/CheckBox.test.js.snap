// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNCheckBox "
>
  <input
    checked={false}
    className="Active"
    id={1234}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
